orbit/pkg/update/update.go

```diff
@@ -108,9 +116,27 @@ func NewUpdater(opt Options) (*Updater, error) {
 		return nil, errors.New("opt.LocalStore must be non-nil")
 	}
 
-	httpClient := fleethttp.NewClient(fleethttp.WithTLSClientConfig(&tls.Config{
+	tlsConfig := &tls.Config{
 		InsecureSkipVerify: opt.InsecureTransport,
-	}))
+	}
+
+	if opt.ServerCertificate != "" {
+		certs, err := os.ReadFile(opt.ServerCertificate)
+		if err != nil {
+			return nil, fmt.Errorf("reading root CA: %w", err)
+		}
+		rootCAPool := x509.NewCertPool()
+		if ok := rootCAPool.AppendCertsFromPEM(certs); !ok {
+			return nil, errors.New("failed to add certificates to root CA pool")
+		}
```

Comment:
This is implemented by `pkg/certificate.LoadPEM(path)`.

Reply:
Thanks! Addressed in 935894089.